test/intl402/Temporal/Duration/prototype/round/rounding-with-largestunit.js

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2024 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-temporal.duration.prototype.round
+description: Rounding mode is taken into account when largestUnit is present.
+includes: [temporalHelpers.js]
+features: [Temporal]
+---*/
+
+// Based on a test case by Adam Shaw
+
+const dur = Temporal.Duration.from({"hours": 13});
+const zdt = Temporal.ZonedDateTime.from('2024-03-10T00:00:00[America/New_York]');
```

Comment:
```suggestion
const dur = new Temporal.Duration.from(0, 0, 0, 0, /* hours = */ 13, 0, 0, 0, 0, 0);
const zdt = new Temporal.ZonedDateTime(0n, "UTC");
```

(I think the actual contents of the ZonedDateTime don't matter here, so let's just replace it with epoch time 0 and UTC. We should avoid requiring named time zones in non-Intl tests for the time being.)

Reply:
I verified that `0n, "UTC"` still exercises the same code path that had the implementation bug, so I'll go ahead and merge it with this modification.